folly/AtomicHashArray.h

```diff
@@ -128,13 +128,28 @@ class AtomicHashArray : boost::noncopyable {
     int    entryCountThreadCacheSize;
     size_t capacity; // if positive, overrides maxLoadFactor
 
-    constexpr Config() : emptyKey((KeyT)-1),
-                         lockedKey((KeyT)-2),
-                         erasedKey((KeyT)-3),
-                         maxLoadFactor(0.8),
-                         growthFactor(-1),
-                         entryCountThreadCacheSize(1000),
-                         capacity(0) {}
+#ifdef MSVC_NO_CONSTEXPR_EASY_POINTER_CASTS
```

Comment:
I can certainly do that. My goal with the port was to make as few changes as possible to the code that effects more than just MSVC, which is why I only enabled it for MSVC at the moment.